spec/features/admin/orders_spec.rb

```diff
@@ -38,6 +38,36 @@
                   visible: :all)[:innerHTML]).to have_content(/.*Four.*Three.*Two/m)
     end
 
+    scenario "filter by multiple order cycles" do
+      # Set up 3 orders and 3 order cycles
+      order_cycle_2 = create(:simple_order_cycle, name: 'Two')
+      order_cycle_3 = create(:simple_order_cycle, name: 'Three')
+      order_cycle_4 = create(:simple_order_cycle, name: 'Four')
+
+      order_2 = create(:order_with_credit_payment, user: user, distributor: distributor,
+                                                  order_cycle: order_cycle_2)
+      order_3 = create(:order_with_credit_payment, user: user, distributor: distributor,
+                                                  order_cycle: order_cycle_3)
+      order_4 = create(:order_with_credit_payment, user: user, distributor: distributor,
+                                                  order_cycle: order_cycle_4)
+
+      quick_login_as_admin
+      visit 'admin/orders'
+
+      # Select order cycle 2 and 3
+      multi_select2_select 'Two', from: 'q_order_cycle_id_in'
+      multi_select2_select 'Three', from: 'q_order_cycle_id_in'
+
+      # click link "Filter Results"
+      page.find('.filter-actions .button.icon-search').click
+
+      # Order 2 and 3 should show, but not 4
+      expect(page).to have_content order_2.number
```

Comment:
So this spec fails in this commit and then passes with your fixing commit, is that right?

Reply:
Yep that's right, order 2 was not showing, as it was only showing orders from the last selected order cycle. I kept separate commits in case anyone needed to confirm the failing test, but will make sure to commit together in the future.

Reply:
No, separate is fine but make the spec pass in every commit (change it twice).